### DISCUSSION TITLE
Error when using apache + mod_auth_openidc

### DISCUSSION THREAD

**Discussion by patbaie974** [OpenIDC/mod_auth_openidc#1007] [Q&A]
Hi all,

I'm beginner with apache http and mod_auth_openidc.

I had a apache proxy that used LDAPS to control the login of a user.
I changed the apache proxy to change LDAPS by auth_openidc but I have a problem with the OIDCRedirectURI and with OIDCProviderMetadataURL.
When I try with OIDCProviderMetadataURL I have always a time out so I tried with all the endPoint like :

  OIDCProviderIssuer "https://auth.tech.exemple"
  OIDCProviderJwksUri "https://auth.tech.exemple/oauth2/jwks"
  OIDCProviderAuthorizationEndpoint "https://auth.tech.exemple/oauth2/authorize"
  OIDCProviderUserInfoEndpoint "https://auth.tech.exemple/oauth2/userinfo"
  OIDCProviderTokenEndpoint "https://auth.tech.exemple/oauth2/token"

This configuration works except OIDCProviderTokenEndpoint with a timeout also.
I don't understand why OIDCProviderJwksUri "https://auth.tech.exemple/oauth2/jwks" works but not OIDCProviderTokenEndpoint "https://auth.tech.exemple/oauth2/token".

So when I try with this configuration I'm redirected to the authentication server, I put my login/pwd, I'm redirected to the url set in 
OIDCRedirectURI ("https://moncrm.test.fr/protected/callback"), but after I have this error 

              **Error:
              OpenID Connect Provider error: Error in handling response type.**

In attachment you'll find all the log 

[log_oidc.txt](https://github.com/zmartzone/mod_auth_openidc/files/10608861/log_oidc.txt)

Here you can find the urls that I see in my browser 

https://auth.tech.exemple/oauth2/authorize?response_type=code&scope=openid%20email%20profile&client_id=xxxxx&state=vI9BmzUveFW5FFn4nkY4G2u2DmM&redirect_uri=https%3A%2F%2Fmoncrm.exemple.fr%2Fprotected%2Fcallback&nonce=JBlQFnN0xwg8IfM_bF8fI7qfeDHQGAX7SkApwZgSh84

https://moncrm.exemple.fr/protected/callback?code=3a9ac45d6feeadc2e70aa0f8776ed48f&state=vI9BmzUveFW5FFn4nkY4G2u2DmM&session_state=znNmaSVzB7qYd70xHUAlZSCR%2Fww9kqmtntFfBJL3u1Y%3D.MFVXaE8yTTZ5dHhpeWRhYVlGeHlyWTc2Z1dqOUtUNmJnSDZ6M3o1TTZWNyt3MUhnZW1yK0M0SlE1dU9Jek1RTlpTSmZ3OE9WYUszVEdYcEVsUXJBT3lHRHR0RXBQRld0ZVZKTVZnZWExOWR6UjZGcVIxVURrY3gxb1VxZWlxWXQ

Here you can find the apache configuration 

ProxyRequests off
ProxyPreserveHost on
ProxyTimeout 120

VirtualHost *:443
  LogLevel debug
  ServerName moncrm.exemple.fr

  SSLEngine on
  SSLCipherSuite xxxx
  SSLProtocol All -SSLv2 -SSLv3
  SSLHonorCipherOrder On
  SSLCompression off
  SSLCertificateFile xxxxxxx
  SSLCertificateKeyFile xxxxxx
  SSLProxyEngine on
  SSLProxyCheckPeerExpire off
  SSLProxyVerify none
  SSLProxyCheckPeerCN off
  SSLProxyCheckPeerName off

  OIDCHTTPTimeoutShort 60
"#"  OIDCProviderMetadataURL https://auth.tech.exemple/.well-known/openid-configuration
  OIDCProviderIssuer "https://auth.tech.exemple"
  OIDCProviderJwksUri "https://auth.tech.exemple/oauth2/jwks"
  OIDCProviderAuthorizationEndpoint "https://auth.tech.exemple/oauth2/authorize"
  OIDCProviderUserInfoEndpoint "https://auth.tech.exemple/oauth2/userinfo"
  OIDCProviderTokenEndpoint "https://auth.tech.exemple/oauth2/token"
  OIDCCookieDomain francetelecom.fr

  OIDCSSLValidateServer Off
  OIDCClientID "xxxxx"
  OIDCClientSecret "xxxxxxxx"
  OIDCScope "openid email profile"
  OIDCRedirectURI "https://moncrm.exemple.fr/protected/callback"
  OIDCCryptoPassphrase "test"
  OIDCResponseType "code"

  ProxyPass /csrt  https://moncrm.cie.exemple.fr/csrt
  ProxyPassReverse /csrt https://moncrm.cie.exemple.fr/csrt

  Location
      AuthType openid-connect
      SetEnv proxy-chain-auth
      Require valid-user
  Location
VirtualHost

Thanks a lot for your help

BR

**Comment by zandbelt** [comment]
```
[Sat Feb 04 18:21:23.601697 2023] [auth_openidc:error] [pid 42476] [client 10.115.64.19:57208] oidc_util_http_call: curl_easy_perform() failed on: https://auth.tech.exemple/oauth2/token (Connection timed out after 60001 milliseconds), referer: https://moncrm.exemple.fr/
```
suggest a firewall or network issue between the Apache server and the Provider

**Reply by patbaie974** [reply]
Yes I think that's the problem but why only for the well-known et token url but not the other?
Do you know how I can fix the proxy to use ? with HTTP_PROXY/HTTPS_PROX and NO_PROXY for some URL. Can I put this configuration in the apache httpd ?

**Reply by zandbelt** [reply]
https://github.com/zmartzone/mod_auth_openidc/wiki#17-restricted-web-server-environments

**Reply by patbaie974** [reply]
Thanks a lot for your help
